packages/override_api_endpoint/lib/src/override_api_endpoint.dart

```diff
@@ -0,0 +1,47 @@
+import 'package:shared_preferences/shared_preferences.dart';
+
+typedef GetUri = Future<Uri?> Function();
```

Comment:
I'd strongly suggest either changing the name of that typedef or getting rid of it at all. We can also use FutureOr instead of Future.